inc/Editor/DataBinding/BlockBindings.php

```diff
@@ -198,6 +198,32 @@ private static function execute_queries( array $block_context, array $source_arg
 		}
 	}
 
+	public static function should_render_empty_result( WP_Block $block ): bool {
+		$block_context = $block->context[ self::$context_name ] ?? [];
+
+		// Only give back true if there are no results
+		return isset( $block_context['results'] ) && empty( $block_context['results'] );
+	}
+
+	public static function get_empty_result_message( WP_Block $block ): ?string {
+		$block_context = $block->context[ self::$context_name ] ?? [];
+		$empty_message = __( 'No results found.', 'remote-data-blocks' );
+
+		/**
+		 * Filter the empty result message for a block binding.
+		 *
+		 * @param string $empty_message The original empty result message.
+		 * @param array $block_context The block context.
+		 */
+		$empty_message = apply_filters(
```

Comment:
This allows the empty message to be customized, in case users want to tweak this. Note that, this is meant to be just a string rather than an HTML that could create problems for us.